<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Setup Instructions</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<!-- Font Awesome -->
<!-- <link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" /> -->
<!-- <script src="https://use.fontawesome.com/54ee8c2dfd.js"></script> -->

<!-- Google fonts -->
<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,400italic,700,700italic|Oswald:400,700' rel='stylesheet' type='text/css'>

<!-- Favicon -->
<link rel="shortcut icon" type="image/x-icon" href="img/favicon.ico">

<!-- Google analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-8298649-8', 'auto');
  ga('send', 'pageview');
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" type="text/css" />
<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ICoR2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="setup.html">
    <span class="fa fa-cog"></span>
     
    Setup
  </a>
</li>
<li>
  <a href="data.html">
    <span class="fa fa-download"></span>
     
    Data
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Lessons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">--- Core Lessons ---</li>
    <li>
      <a href="r-basics.html">R Basics</a>
    </li>
    <li>
      <a href="r-dataframes.html">Data Frames</a>
    </li>
    <li>
      <a href="r-dplyr-yeast.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-tidy.html">Tidying data</a>
    </li>
    <li>
      <a href="r-rmarkdown.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-viz-gapminder.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-ggtree.html">Phylogenetic Trees</a>
    </li>
    <li>
      <a href="r-rnaseq-airway.html">RNA-seq</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">--- Optional Lessons ---</li>
    <li>
      <a href="r-stats.html">Essential Statistics with R</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">--- Homework ---</li>
    <li>
      <a href="r-dplyr-homework.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-dataviz-homework.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-rmarkdown-homework.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-stats-homework.html">Essential Statistics</a>
    </li>
    <li>
      <a href="r-rnaseq-homework.html">RNA-seq Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-question fa-lg"></span>
     
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index.html#faq">FAQ</a>
    </li>
    <li>
      <a href="help.html">Further resources</a>
    </li>
    <li>
      <a href="https://github.com/4va/biodatasci">Source code for this site</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Setup Instructions</h1>

</div>


<div id="software" class="section level2">
<h2>Software</h2>
<div id="r" class="section level3">
<h3>R</h3>
<p><strong>Install R.</strong> You’ll need R version <strong>3.4.0</strong> or higher.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> Download and install R for <a href="http://cran.r-project.org/bin/windows/base/">Windows</a> or <a href="http://cran.r-project.org/bin/macosx/">Mac</a> (download the latest R-3.x.x.pkg file for your appropriate version of Mac OS).</p>
</div>
<div id="rstudio" class="section level3">
<h3>RStudio</h3>
<p>Download and install <strong><a href="https://www.rstudio.com/products/rstudio/download/">RStudio Desktop</a></strong> version &gt;= 1.0.143.</p>
<p>R and RStudio are separate downloads and installations. <strong>R</strong> is the underlying statistical computing environment, but using R alone is no fun. <strong>RStudio</strong> is a graphical integrated development environment that makes using R much easier. You need R installed before you install RStudio.</p>
</div>
<div id="cran-packages" class="section level3">
<h3>CRAN packages</h3>
<p>Launch RStudio (RStudio, <em>not R itself</em>). Ensure that you have internet access, then copy and paste the following command into the <strong>Console</strong> panel (usually the lower-left panel, by default) and hit the Enter/Return key.</p>
<pre class="r"><code>install.packages(&quot;tidyverse&quot;)</code></pre>
<p><em>A few notes</em>:</p>
<ul>
<li>Commands are case-sensitive.</li>
<li>You must be connected to the internet.</li>
<li>The <a href="http://tidyverse.org/">tidyverse</a> package is kind of a meta-package that automatically installs/loads many core packages that we use throughout the workshops.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></li>
<li>Even if you’ve installed these packages in the past, do re-install the most recent version. Many of these packages are updated often, and we may use new features in the workshop that aren’t available in older versions.</li>
<li>If you’re using Windows you might see errors about not having permission to modify the existing libraries – disregard these. You can avoid this by running RStudio as an administrator (right click the RStudio icon, then click “Run as Administrator”).</li>
</ul>
<p>Check that you’ve installed everything correctly by closing and reopening RStudio and entering the following commands at the console window (don’t worry about the <em>Conflicts with tidy packages</em> warning):</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<p>This may produce some notes or other output, but as long as you don’t get an error message, you’re good to go. If you get a message that says something like: <code>Error in library(somePackageName) : there is no package called 'somePackageName'</code>, then the required packages did not install correctly. Please do not hesitate to email me <em>prior to the course</em> if you are still having difficulty. In this email, please copy and paste what you typed in the console, and all of the output that streams by in the console.</p>
</div>
<div id="bioconductor" class="section level3">
<h3>Bioconductor</h3>
<p>For some classes (e.g., RNA-seq), you’ll need to install a few <a href="http://bioconductor.org/">Bioconductor</a> packages. These packages are installed differently than “regular” R packages from CRAN. Copy and paste these lines of code into your R console <strong>one at a time</strong>.</p>
<pre class="r"><code>source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite()
biocLite(&quot;DESeq2&quot;)
biocLite(&quot;ggtree&quot;)</code></pre>
<p>A few notes:</p>
<ul>
<li>We will be using the latest versions of Bioconductor from the 3.5 release. This requires R version 3.4.0 or higher. If you have R 3.4.0 installed, running the commands above will install Bioconductor 3.5. See <a href="http://bioconductor.org/news/bioc_3_5_release/" class="uri">http://bioconductor.org/news/bioc_3_5_release/</a>.</li>
<li>If at any point in the Bioconductor package installations you get prompts in the console asking you to update any existing packages, type <code>n</code> at the prompt at hit enter.</li>
<li>If you see a note long the lines of “<em>binary version available but the source version is later</em>”, followed by a question, “<em>Do you want to install from sources the package which needs compilation? y/n</em>”, type <strong><code>n</code> for no</strong>, and hit enter.</li>
</ul>
<p>Check that you’ve installed everything correctly by closing and reopening RStudio and entering the following commands at the console window, one at a time:</p>
<pre class="r"><code>library(DESeq2)
library(ggtree)</code></pre>
<p>If you get a message that says something like: <code>Error in library(somePackageName) : there is no package called 'somePackageName'</code>, then the required packages did not install correctly. Please do not hesitate to email me <em>prior to the course</em> if you are still having difficulty. In this email, please copy and paste what you typed in the console, and all of the output that streams by in the console.</p>
</div>
<div id="rmarkdown" class="section level3">
<h3>RMarkdown</h3>
<p>Several additional setup steps required for the reproducible research with RMarkdown class.</p>
<ol style="list-style-type: decimal">
<li>First, install R, RStudio, and the tidyverse package as described above. Also install the knitr and rmarkdown packages.</li>
</ol>
<pre class="r"><code>install.packages(&quot;knitr&quot;)
install.packages(&quot;rmarkdown&quot;)</code></pre>
<ol style="list-style-type: decimal">
<li>Next, launch RStudio (not R). Click File, New File, R Markdown. This may tell you that you need to install additional packages (<code>knitr</code>, <code>yaml</code>, <code>htmltools</code>, <code>caTools</code>, <code>bitops</code>, <code>rmarkdown</code>, and maybe a few others). Click “Yes” to install these.</li>
<li><strong><em>Optional</em>:</strong> If you want to convert to PDF, you will need to install a <strong><span class="math inline">\(\LaTeX\)</span></strong> typesetting engine. This differs on Mac and Windows. <em>Note that this part of the installation may take up to several hours, and isn’t strictly required for the class.</em>
<ul>
<li><strong>Windows</strong>: Download and install MiKTeX: <a href="https://miktex.org/download" class="uri">https://miktex.org/download</a>. Read the installation tutorial first at <a href="https://miktex.org/howto/install-miktex" class="uri">https://miktex.org/howto/install-miktex</a>.</li>
<li><strong>Mac</strong>: Download and install MacTeX.pkg at <a href="http://www.tug.org/mactex/mactex-download.html" class="uri">http://www.tug.org/mactex/mactex-download.html</a>. This is a very large download (&gt;2 gigabytes), and may take a while depending on your network speed. Do this <em>prior to the course</em>.</li>
</ul></li>
</ol>
</div>
</div>
<div id="get-data" class="section level2">
<h2>Get Data</h2>
<p>Click the <strong><a href="data.html"><i class="fa fa-download"></i> Data</a></strong> link on the navbar at the top. You can download all the data needed by downloading <strong><a href="data.zip">this zip file <i class="fa fa-file-archive-o" aria-hidden="true"></i></a></strong> or by downloading individual data sets as needed at the <strong><a href="data.html"><i class="fa fa-download"></i> Data</a></strong> page.</p>
</div>
<div id="required-reading" class="section level2">
<h2>Required Reading</h2>
<div id="tidy-data" class="section level3">
<h3>Tidy Data</h3>
<ol style="list-style-type: decimal">
<li>“For Big-Data Scientists, Janitor Work Is Key Hurdle to Insights.” <em>The New York Times</em>, August 8, 2014. Available at: <a href="https://nyti.ms/2jNUfHo" class="uri">https://nyti.ms/2jNUfHo</a>.</li>
<li>Sections 1-3 of <a href="http://www.jstatsoft.org/article/view/v059i10/v59i10.pdf">Wickham, H. “Tidy Data.” <em>Journal of Statistical Software</em> 59:10 (2014)</a>.</li>
<li>Click through each section of <a href="http://stephenturner.us/dataorganization/">stephenturner.us/dataorganization/</a>.</li>
</ol>
</div>
<div id="rmarkdown-background" class="section level3">
<h3>RMarkdown Background</h3>
<p>Spend a few minutes to learn a little bit about <em>Markdown</em>. All you really need to know is that Markdown is a lightweight markup language that lets you create styled text (like <strong>bold</strong>, <em>italics</em>, <a href="http://bioconnector.org/markdown">links</a>, etc.) using a very lightweight plain-text syntax: <code>(like **bold**, _italics_, [links](http://bioconnector.org/markdown), etc.)</code>. The resulting text file can be <em>rendered</em> into many downstream formats, like PDF (for printing) or HTML (websites).</p>
<ol style="list-style-type: decimal">
<li><em>(30 seconds)</em> Read the summary paragraph on the <a href="https://en.wikipedia.org/wiki/Markdown">Wikipedia page</a>.</li>
<li><em>(1 minute)</em> Bookmark and refer to this markdown reference: <a href="http://commonmark.org/help/" class="uri">http://commonmark.org/help/</a>.</li>
<li><em>(5-10 minutes)</em> Run through this 10-minute in-browser markdown tutorial: <a href="http://commonmark.org/help/tutorial/" class="uri">http://commonmark.org/help/tutorial/</a>.</li>
<li><em>(5-10 minutes)</em> Go to <a href="http://dillinger.io/" class="uri">http://dillinger.io/</a>, an in-browser Markdown editor, and play around. Write a simple markdown document, and export it to HTML and/or PDF.</li>
<li><em>(10 minutes)</em> See RStudio’s excellent documentation on Rmarkdown at <a href="http://rmarkdown.rstudio.com/" class="uri">http://rmarkdown.rstudio.com/</a>. Click “Getting Started” and watch the 1 minute video on the <a href="http://rmarkdown.rstudio.com/lesson-1.html">Introduction page</a>. Continue reading through each section here on the navigation bar to the left (<em>Introduction</em> through <em>Cheatsheets</em>, and optionally download and print out the cheat sheet). Finally, browse through the <a href="http://rmarkdown.rstudio.com/gallery.html">RMarkdown Gallery</a>.</li>
<li><em>(0 seconds)</em> No need to look now, but don’t forget that the course <a href="help.html">help page</a> has some <a href="help.html#markdown-rmarkdown-resources">useful resources on Markdown+RMarkdown</a>.</li>
</ol>
</div>
<div id="phylogenetic-trees" class="section level3">
<h3>Phylogenetic trees</h3>
<ol style="list-style-type: decimal">
<li>How to read a phylogeny <a href="http://epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny">epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny</a>.</li>
</ol>
</div>
<div id="rna-seq" class="section level3">
<h3>RNA-seq</h3>
<ol style="list-style-type: decimal">
<li><a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8">Conesa et al. A survey of best practices for RNA-seq data analysis. <em>Genome Biology</em> 17:13 (2016)</a>.</li>
<li><a href="https://f1000research.com/articles/4-1521/v2">Soneson et al. “Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences.” <em>F1000Research</em> 4 (2015)</a>.</li>
<li>Abstract and introduction sections of <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625">Himes et al. “RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells.” <em>PLoS ONE</em> 9.6 (2014): e99625</a>.</li>
</ol>
<hr />
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>R version 3.4.0 was released April 2017. If you have not updated your R installation since then, you need to upgrade to a more recent version, since several of the required packages depend on a version at least this recent. You can check your R version with the <code>sessionInfo()</code> command.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Installing/loading the tidyverse <strong>tidyverse</strong> will install/load the core tidyverse packages that you are likely to use in almost every analysis: <strong>ggplot2</strong> (for data visualisation), <strong>dplyr</strong> (for data manipulation), <strong>tidyr</strong> (for data tidying), <strong>readr</strong> (for data import), <strong>purrr</strong> (for functional programming), and <strong>tibble</strong> (for tibbles, a modern re-imagining of data frames). It also installs a selection of other tidyverse packages that you’re likely to use frequently, but probably not in every analysis (these are installed, but you’ll have to load them separately with <code>library(packageName)</code>). This includes: <strong>hms</strong> (for times), <strong>stringr</strong> (for strings), <strong>lubridate</strong> (for date/times), <strong>forcats</strong> (for factors), <strong>DBI</strong> (for databases), <strong>haven</strong> (for SPSS, SAS and Stata files), <strong>httr</strong> (for web apis), <strong>jsonlite</strong> (or JSON), <strong>readxl</strong> (for .xls and .xlsx files), <strong>rvest</strong> (for web scraping), <strong>xml2</strong> (for XML), <strong>modelr</strong> (for modelling within a pipeline), and <strong>broom</strong> (for turning models into tidy data). After installing tidyverse with <code>install.packages(&quot;tidyverse&quot;)</code> and loading it with <code>library(tidyverse)</code>, you can use <code>tidyverse_update()</code> to update all the tidyverse packages installed on your system at once.<a href="#fnref2" class="footnote-back">↩</a></p></li>
</ol>
</div>

<div class="footer">
<a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank"><i class="fa fa-creative-commons fa-2x"></i></a>&nbsp;
<a href="https://twitter.com/genetics_blog" target="_blank"><i class="fa fa-twitter fa-2x"></i></a>&nbsp;
<a href="https://github.com/4va/biodatasci" target="_blank"><i class="fa fa-github fa-2x"></i></a>&nbsp;
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
